go/consensus/cometbft/apps/roothash/state/state.go

```diff
@@ -303,6 +374,82 @@ func (s *MutableState) SetRuntimeState(ctx context.Context, state *roothash.Runt
 	if err := s.ms.Insert(ctx, ioRootKeyFmt.Encode(&state.Runtime.ID), ioRoot); err != nil {
 		return api.UnavailableStateError(err)
 	}
+
+	// Clean previously stored state and I/O roots if we're over the maximum.
+	params, err := s.ConsensusParameters(ctx)
+	var maxStored uint32
+	if err == nil {
+		maxStored = params.MaxPastRootsStored
+	}
```

Comment:
The `ConsensusParameters` method failing is a critical error and should not happen in practice.
```suggestion
	params, err := s.ConsensusParameters(ctx)
	if err != nil {
		return api.UnavailableStateError(err)
	}
	maxStored := params.MaxPastRootsStored
```